pkg/secrets/check_rights_windows.go

```diff
@@ -53,34 +47,38 @@ func checkRights(filename string, allowGroupExec bool) error {
 
 	// create the sids that are acceptable to us (local system account and
 	// administrators group)
-	var localSystem *windows.SID
-	err = windows.AllocateAndInitializeSid(&windows.SECURITY_NT_AUTHORITY,
-		1, // local system has 1 valid subauth
-		windows.SECURITY_LOCAL_SYSTEM_RID,
-		0, 0, 0, 0, 0, 0, 0,
-		&localSystem)
+	localSystem, err := getLocalSystemSID()
 	if err != nil {
 		return fmt.Errorf("could not query Local System SID: %s", err)
 	}
 	defer windows.FreeSid(localSystem)
 
-	var administrators *windows.SID
-	err = windows.AllocateAndInitializeSid(&windows.SECURITY_NT_AUTHORITY,
-		2, // administrators group has 2 valid subauths
-		windows.SECURITY_BUILTIN_DOMAIN_RID,
-		windows.DOMAIN_ALIAS_RID_ADMINS,
-		0, 0, 0, 0, 0, 0,
-		&administrators)
+	administrators, err := getAdministratorsSID()
 	if err != nil {
 		return fmt.Errorf("could not query Administrator SID: %s", err)
 	}
 	defer windows.FreeSid(administrators)
 
-	secretuser, err := winutil.GetSidFromUser()
+	currentUser, err := winutil.GetSidFromUser()
 	if err != nil {
 		return fmt.Errorf("could not get SID for current user: %s", err)
 	}
 
+	secretuser := currentUser
+
+	elevated, err := winutil.IsProcessElevated()
+	if err != nil {
+		return fmt.Errorf("unable to determine if running elevated: %s", err)
```

Comment:
Are we unable to determine if we're elevated or we know that we are not elevated?

Reply:
"unable to determine if we're elevated"